backend/instancemgmt/instance_manager_test.go

```diff
@@ -204,13 +218,13 @@ func TestInstanceManagerConcurrency(t *testing.T) {
 type testInstance struct {
 	orgID         int64
 	updated       time.Time
-	disposed      bool
-	disposedTimes int64
+	disposed      atomic.Bool
+	disposedTimes atomic.Int64
```

Comment:
So I had to change these since race detection complained about these due to concurrent reads/write in tests. In my head, given that we dispose within `im.locker.Lock(cacheKey)`, there should only ever be one goroutine doing the creation of a new instance and calling Dispose hency why the instance itself shouldn't have to do any additional locking in their Dispose method.